runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,357 @@
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    boolean hasNullOrEmptyOrDuplicates =
+                            remoteDestinations.stream().anyMatch(Objects::isNull) ||
+                            remoteDestinations.stream().anyMatch(String::isEmpty) ||
+                            remoteDestinations.stream().distinct().count() < remoteDestinations.size();
+
+                    if (hasNullOrEmptyOrDuplicates) {
+                        throw new IllegalArgumentException("Invalid destination(s) found in list");
+                    }
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        DestinationInfoVal clientDestinationIds = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+                        List<String> clientDestinationIdsList = new ArrayList<>(clientDestinationIds.getDestinationIdsList());
+
+                        int initialNumDestinations = clientDestinationIdsList.size();
+                        clientDestinationIdsList.removeAll(remoteDestinations);
+
+                        if (clientDestinationIdsList.size() == initialNumDestinations) {
+                            throw new NoSuchElementException("No matching destinations found");
+                        } else if (clientDestinationIdsList.size() + remoteDestinations.size() > initialNumDestinations) {
+                            log.warn("Not all destinations to remove are present");
+                        }
+
+                        if (clientDestinationIdsList.size() == 0) {
+                            txn.delete(sourceMetadataTable, clientInfoKey);
+                        } else {
+                            clientDestinationIds = clientDestinationIds.toBuilder()
+                                    .clearDestinationIds()
+                                    .addAllDestinationIds(clientDestinationIdsList)
+                                    .build();
+                            txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinationIds, null);
+                        }
+                        txn.commit();
+
+                        return null;
+                    }
+                    throw new NoSuchElementException(String.format("Record not found for group: %s", logicalGroup));
+                } catch (TransactionAbortedException tae) {
+                    log.error("Unable to remove destinations, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Unable to remove destinations");
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    private static boolean isValid(final Object obj) {
```

Comment:
any reason this method is static?

Reply:
it doesn't depend on any instance vars so figured static was fine in this case.